debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SourceTimestampMode.java

```diff
@@ -17,12 +21,44 @@
     /**
      * This mode (default) will set the source timestamp field (ts_ms) of when the record was committed in the database.
      */
-    COMMIT("commit"),
+    COMMIT("commit") {
+        @Override
+        protected Instant getTimestamp(SqlServerConnection connection, ResultSet resultSet, Clock clock) throws SQLException {
```

Comment:
At another iteration, this method's signature could be simplified to `getTimestamp(ResultSet resultSet)`. The connection and clock could be replaced with constructor arguments for the `COMMIT` and `PROCESSING` values respectively.